sync/sync.go

```diff
@@ -617,3 +628,121 @@ func (t *Task) getToReplica(address string) (rest.Replica, error) {
 
 	return rest.Replica{}, fmt.Errorf("Failed to find target replica to copy to")
 }
+
+// InternalSnapshotCleaner should be run in the background, it tries to delete a snapshot every 60s
+// It fetches the checkpoint from controller which is present at in-memory of controller.
+// If checkpoint is not available at controller, snapshot delete is not initiated.
+// A deletion candidate list is prepared
+// In each iteration of the loop, one snapshot is picked from the top and deleted.
+func (t *Task) InternalSnapshotCleaner(s *replica.Server, repClient *replicaClient.ReplicaClient) {
+	ticker := time.NewTicker(SnapshotDeletionInterval)
+
+	for range ticker.C {
+		snapshot, err := t.client.GetCheckpoint()
+		if err == nil && snapshot == s.Replica().Info().Checkpoint {
+			sortedSnapshotList, _ := getDeleteCandidateChain(s, snapshot)
+			if len(sortedSnapshotList) < 10 {
+				continue
+			}
+			var snap SnapList
+			for _, snap = range sortedSnapshotList {
+				if snap.name != "" {
```

Comment:
can snap.name possible to be empty?

Reply:
It cannot be empty over here. This has been added just as a precausion